src/webapi/indexeddb.rs

```diff
@@ -0,0 +1,1145 @@
+                return 2;
+            } else {
+                return 3;
+            }
+        }.try_into().unwrap();
+        match t {
+            0 => IDBRequestSource::Store(js!(return @{self.as_ref()}.source;).try_into().unwrap()),
+            1 => IDBRequestSource::Index(js!(return @{self.as_ref()}.source;).try_into().unwrap()),
+            2 => IDBRequestSource::Cursor(js!(return @{self.as_ref()}.source;).try_into().unwrap()),
+            3 => IDBRequestSource::None,
+            _ => panic!()
+        }
+    }
+    
+    /// The `transaction` read-only property of the `IDBRequest` interface
+    /// returns the transaction for the request, that is, the transaction
+    /// the request is being made inside.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest/transaction)
+    fn transaction( &self ) -> Option<IDBTransaction> {
+        let transaction : Value = js! (
+            return @{self.as_ref()}.transaction;
+        );
+        match transaction {
+            Value::Undefined => None,
+            Value::Null => None,
+            _ => Some(transaction.try_into().unwrap())
+        }
+    }
+    
+    /// The `ready_state` read-only property of the `IDBRequest` interface
+    /// returns the state of the request.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest/readyState)
+    fn ready_state( &self ) -> IDBRequestReadyState {
+        let ready_state : String = js! (
+            return @{self.as_ref()}.readyState;
+        ).try_into().unwrap();
+        
+        if ready_state.eq("pending") {
+            return IDBRequestReadyState::Pending;
+        } else if ready_state.eq("done") {
+            return IDBRequestReadyState::Done;
+        } else {
+            panic!("Got {} as an IDBRequestReadyState.", ready_state);
+        }        
+    }
+    
+}
+
+/// The `IDBReques`t interface of the IndexedDB API provides access to results
+/// of asynchronous requests to databases and database objects using event
+/// handlers. Events that are received are IDBSuccessEvent and IDBErrorEvent.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest)
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "IDBRequest")]
+#[reference(subclass_of(EventTarget))]
+pub struct IDBRequest( Reference );
+
+impl IEventTarget for IDBRequest {}
+impl IDBRequestSharedMethods for IDBRequest {}
+
+/// Provides access to the results of requests to open or delete databases.
+/// Receives `IDBBlockedEvent` and `IDBVersionChangeEvent` as well as events received by `IDBRequest`.
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBOpenDBRequest)
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "IDBOpenDBRequest")]
+#[reference(subclass_of(EventTarget))]
+pub struct IDBOpenDBRequest( Reference );
```

Comment:
Trying to fix #2 this using subclass_of:

```diff
diff --git a/src/lib.rs b/src/lib.rs
index b0b56c5..464a713 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -241,7 +241,6 @@ pub mod web {
         pub use webapi::indexeddb::{
             IDBOpenDBRequest,
             IDBDatabase,
-            IDBRequestSharedMethods,
             IDBRequest,
             IDBIndex,
             IDBObjectStore,
diff --git a/src/webapi/indexeddb.rs b/src/webapi/indexeddb.rs
index cad336e..5100ad5 100644
--- a/src/webapi/indexeddb.rs
+++ b/src/webapi/indexeddb.rs
@@ -26,10 +26,18 @@ pub enum IDBRequestSource {
     Cursor(IDBCursor)
 }
 
-/// IDBRequestSharedMethode represents the methode that are shared between
-/// IDBOpenDBRequest and IDBRequest.
+/// The `IDBRequest` interface of the IndexedDB API provides access to results
+/// of asynchronous requests to databases and database objects using event
+/// handlers. Events that are received are IDBSuccessEvent and IDBErrorEvent.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest)
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "IDBRequest")]
+#[reference(subclass_of(EventTarget))]
+pub struct IDBRequest( Reference );
+
 // https://w3c.github.io/IndexedDB/#idbrequest
-pub trait IDBRequestSharedMethods : IEventTarget {
+impl IDBRequest {
 
     /// The result read-only property of the `IDBRequest` interface returns the result of the request,
     /// or if the request failed InvalidStateError.
@@ -103,30 +111,16 @@ pub trait IDBRequestSharedMethods : IEventTarget {
     
 }
 
-/// The `IDBRequest` interface of the IndexedDB API provides access to results
-/// of asynchronous requests to databases and database objects using event
-/// handlers. Events that are received are IDBSuccessEvent and IDBErrorEvent.
-///
-/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest)
-#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
-#[reference(instance_of = "IDBRequest")]
-#[reference(subclass_of(EventTarget))]
-pub struct IDBRequest( Reference );
-
 impl IEventTarget for IDBRequest {}
-impl IDBRequestSharedMethods for IDBRequest {}
 
 /// Provides access to the results of requests to open or delete databases.
 /// Receives `IDBBlockedEvent` and `IDBVersionChangeEvent` as well as events received by `IDBRequest`.
 /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBOpenDBRequest)
 #[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
 #[reference(instance_of = "IDBOpenDBRequest")]
-#[reference(subclass_of(EventTarget))]
+#[reference(subclass_of(IDBRequest))]
 pub struct IDBOpenDBRequest( Reference );
 
-impl IEventTarget for IDBOpenDBRequest {}
-impl IDBRequestSharedMethods for IDBOpenDBRequest {}
-
 impl IDBOpenDBRequest {
 
     /// Returns the value property as an `IDBDatabase`, or an `InvalidStateError`.
```

but it can't find the method:

```
stdweb $ (cd examples/indexeddb/; cargo web start --target=wasm32-unknown-unknown)
   Compiling stdweb v0.4.17 (stdweb)
error[E0599]: no method named `result` found for type `&webapi::indexeddb::IDBOpenDBRequest` in the current scope
   --> src/webapi/indexeddb.rs:128:20
    |
128 |         match self.result() {
    |                    ^^^^^^
``

any idea how to best address this? 

As for #1 (the name containing a second "db"),  this helps clarify the method opens a database as opposed to a cursor or key, and OpenDB matches the IDL nomenclature...could be changed, though.